res/controllers/midi-components-0.0.js

```diff
@@ -467,9 +467,19 @@
             // For the first messages, disregard the LSB in case
             // the first LSB is received after the first MSB.
             if (this.MSB === undefined) {
-                this.max = 127;
+                // a flaw in the Pot API does not mandate consumers
+                // to supply an accurate max value when using
+                // this with non-7-bit inputs. We cannot detect
+                // that in the constructor so set the max
+                // appropriately here
+                if (this.max === Component.prototype.max) {
+                    this.max = (1 << 14) - 1;
+                }
+                var maxSaved = this.max;
+                // maximum value of MSB
+                this.max = this.max >> 7;
                 this.input(channel, control, value, status, group);
```

Comment:
Temporary change a member to before calling a function looks hacky. 

Why is this code required? Is this an initial call with low precision? 
  
Let's say we have 8 bit precision, than the MSB is 1 for the max value.
inValueScale() has div by 0 with that and will not work well with other small bit counts.

Can this call be removed? 

Alternative we may leave max untouched just set the this.MSB before and pass 0 as value. 

By the way, every input function does 
                    ```
                    if (this.MSB !== undefined) {
                        value = (this.MSB << 7) + value;
                    }
```
That logic can be moved to inputLSB() so that value is always the truth an the single precision code dos not suffering form the penalty of checking  this.MSB

Reply:
> Temporary change a member to before calling a function looks hacky.

I agree, but the entire `this.MSB !== undefined` branch is a hack... 

> Why is this code required? Is this an initial call with low precision?

Yes. 

> Can this call be removed?

Maybe. I'll look into it.

Reply:
The problem with removing this call (and by extension the entire branch) is that requesting the initial controller state will break if the LSB is received before the MSB.

Reply:
In the C Domain we allow both orders: 
https://github.com/mixxxdj/mixxx/blob/7672cf1a5efcc17b0ead2f28c7585414fea41b7b/src/controllers/midi/midicontroller.cpp#L318
The requirement is that both messages are send with nothing else in between.

Reply:
I'm sorry but I have 0 idea what you are trying to demonstrate with these tables.

I know this code is not technically correct, but for Pot controls it rarely matters.

I can't do much about that because any change requires more information about the hardware that uses this code and thus is not backwards compatible.

Reply:
> In the C Domain we allow both orders:

Yes and you know the order in advance. We don't know that in the JS code because nobody told us and we had no option to guess it either. I guess I could implement some sort of checking in the input handlers, but that will increase complexity by a lot.

Reply:
First of all, we discussing no longer the scope of this PR.
I have not understand the controller state issue, that requires the discussed branch. 

In the C domain we also do not know the order in advance. We collect MSB and LSB messages and when both is there we calculate the value. We discard a stored MSB or LSB when any other message is send. This ensures that both are actually belong together. 

Of cause that can ticked as well, but at least better than always using the wrong order.

Reply:
> First of all, we discussing no longer the scope of this PR.

So whats the point then? I know the design is flawed. So will we agree on that and discuss whats actually in the scope of this PR or will we blow it 10x as we usually do?

> In the C domain we also do not know the order in advance.

Are you sure? Thats what I figured the distinction between `MidiOption::FourteenBitLSB` and `MidiOption::FourteenBitMSB` was. 

> I have not understand the controller state issue, that requires the discussed branch.

Ok going back on topic. We don't know in which order `inputLSB` and `inputMSB` will be called so to basically avoid throwing an error deeper into the callstack we ignore the LSB message and then when we receive the MSB, we use that as the LSB as a single-shot workaround. Does that answer your question?

Reply:
>> First of all, we discussing no longer the scope of this PR.

> So whats the point then? I know the design is flawed. So will we agree on that and discuss whats actually in the scope of this PR or will we blow it 10x as we usually do?

That was my point, not blowing up this PR. 

I have now understand the issue. 

> Are you sure? Thats what I figured the distinction between MidiOption::FourteenBitLSB and MidiOption::FourteenBitMSB was.

Yes, pretty sure. We even have a test for that.

Reply:
So for the solution here, wan may to shift the "value" left, instead of sifting the "max" right. 
If we in addition store the discarded LSB we can put it in and have a fully valid message.

Reply:
Now I understand your proposal. Thanks. I just pushed the commit containing your suggestions.